### PR TITLE
Adding a pattern for SelectOp in the AdjustIntegerWidthPass

### PR DESCRIPTION
**Pull request by harsh-nod**

### CODE REVIEW

**Comment by harsh-nod** [comment]
Hi Hanhan,

As per your suggestions I removed the addition of the OrOp from this PR and also added a test in adjust_integer_width.mlir. Let me know what you think.

Thanks!

File: iree/compiler/Translation/SPIRV/XLAToSPIRV/test/adjust_integer_width.mlir [hunk 1]
```diff
@@ -272,4 +272,26 @@ module{
...
+    spv.globalVariable @globalInvocationID built_in("GlobalInvocationId") : !spv.ptr<vector<3xi32>, Input>
+    spv.globalVariable @constant_arg_0 bind(0, 0) : !spv.ptr<!spv.struct<i1 [0]>, StorageBuffer>
+    spv.globalVariable @constant_arg_1 bind(0, 1) : !spv.ptr<!spv.struct<i8 [0]>, StorageBuffer>
+    spv.func @select(%arg0 : i1, %arg1 : i8) -> () "None" {
+      %0 = spv._address_of @constant_arg_0 : !spv.ptr<!spv.struct<i1 [0]>, StorageBuffer>
+      %1 = spv.constant 0 : i32
+      %2 = spv.AccessChain %0[%1] : !spv.ptr<!spv.struct<i1 [0]>, StorageBuffer>
+      %3 = spv.Load "StorageBuffer" %2 : i1
+      // CHECK: spv.constant 0 : i32
+      // CHECK: spv.constant 1 : i32
```
Comment (hanhanW) [inline iree/compiler/Translation/SPIRV/XLAToSPIRV/test/adjust_integer_width.mlir]: It's quite weird to check these two constant without capturing. In this test case, we are testing if spv.Select get adjusted. There are two options:

1) Check spv.Select itself.
2) Capture the values for spv.constant, and also for loading value, then check spv.Select with them.

For option two, more details can be found at https://mlir.llvm.org/getting_started/TestingGuide/

**Comment by harsh-menon** [comment]
You're right - we should just check spv.Select. I will remove those two checks and upload a new patch.

**Comment by harsh-menon** [comment]
I uploaded a new patch but am not sure why check submodules is failing.

**Comment by ScottTodd** [comment]
> I uploaded a new patch but am not sure why check submodules is failing.

You can ignore that. Looks like we had a little glitch updating the submodules today. Thanks for noticing :)

**Comment by harsh-nod** [comment]
Thanks! Feel free to merge - I dont think I am authorized to merge.

**Pull request closed without merging**